File: solutions/julia/roman-numerals/1/roman-numerals.jl
function to_roman(number)
    an=[1000,900,500,400,100,90,50,40,10,9,5,4,1]
    rn=["M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"]
    if number <= 0
        throw(ErrorException("error"))
    end
    i=1
    r=""
    while number > 0
        if number>=an[i]
            r=r*rn[i]
            number=number-an[i]
        else
            i=i+1
        end
    end
    return r
end

File: solutions/javascript/roman-numerals/1/roman-numerals.js
//
// This is only a SKELETON file for the 'Roman Numerals' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const toRoman = (number) => {
  var rom_cnv = {1000:"M", 900: "CM", 500: "D", 400: "CD",
  100: "C", 90:"XC", 50: "L", 40: "XL",
  10: "X", 9: "IX", 5:"V", 4: "IV", 1:"I"};
  var rev = Object.keys(rom_cnv).reverse();
  var result = "";
  while (number > 0) {
    for (var ki=0; ki<rev.length;ki++){
      var key = rev[ki];
      while (key <= number) {
        var value = rom_cnv[key];
        result += value;
        number = number - key;
      }
    }
  }
  return result;
  //throw new Error('Remove this statement and implement this function');
};
